### PR TITLE
bsc: gas + add asset type

### PR DESCRIPTION
**Pull request by skylarbarrera**
## What changed (plus any additional context for devs)
adding logic to handle bsc gas estimations 

## Screen recordings / screenshots

nothing yet 

## What to test
nothing yet

### CODE REVIEW

File: src/handlers/gasFees.ts [hunk 1]
```diff
@@ -31,3 +31,25 @@ const polygonGasStationApi = new RainbowFetchClient({
...
+  timeout: 30000, // 30 secs
+});
+
+/**
+ * @desc get BSC gas prices
+ * @return {Promise}
+ */
+export const bscGasStationGetGasPrices = () =>
+  bscGasStationApi.get(
+    `?module=gastracker&action=gasoracle&apikey=YourApiKeyToken`
```
Comment (skylarbarrera) [inline src/handlers/gasFees.ts:54]: this is our ghetto way to not get rate limited, see polygon gas api as well

theres a backend ticket to move this over to meteorology and we've decided to keep it until then

File: src/redux/gas.ts [hunk 2]
```diff
@@ -60,6 +62,7 @@ const getGasPricePollingInterval = (network: Network): number => {
     case Network.polygon:
       return 2000;
     case Network.arbitrum:
+    case Network.bsc:
```
Comment (skylarbarrera) [inline src/redux/gas.ts:65]: bsc block times are every 3 seconds

File: src/handlers/gasFees.ts [hunk 3]
```diff
@@ -31,3 +31,25 @@ const polygonGasStationApi = new RainbowFetchClient({
...
+    'Content-Type': 'application/json',
+  },
+  timeout: 30000, // 30 secs
+});
+
+/**
+ * @desc get BSC gas prices
+ * @return {Promise}
+ */
+export const bscGasStationGetGasPrices = () =>
```
Comment (estrattonbailey) [inline src/handlers/gasFees.ts:52]: Do we know the return type/shape? Would be good to type this.

File: src/redux/gas.ts [hunk 4]
```diff
@@ -139,10 +142,19 @@ const getUpdatedGasFeeParams = (
   txNetwork: Network,
   l1GasFeeOptimism: BigNumber | null = null
 ) => {
-  const nativeTokenPriceUnit =
-    txNetwork !== Network.polygon
-      ? ethereumUtils.getEthPriceUnit()
-      : ethereumUtils.getMaticPriceUnit();
+  let nativeTokenPriceUnit = ethereumUtils.getEthPriceUnit();
+
+  switch (txNetwork) {
```
Comment (estrattonbailey) [inline src/redux/gas.ts:147]: Nice `switch`

File: src/redux/gas.ts [hunk 5]
```diff
@@ -281,6 +302,35 @@ const getPolygonGasPrices = async () => {
   return polygonGasStationPrices;
 };
 
+const getBscGasPrices = async () => {
+  const {
+    data: { result },
+  }: {
+    data: GasFeesBscGasStationData;
+  } = await bscGasStationGetGasPrices();
```
Comment (estrattonbailey) [inline src/redux/gas.ts:310]: What happens if there is no result?

Comment (skylarbarrera) [reply src/redux/gas.ts:310]: yeah nice catch, need to add better handling here and with polygon so we can track the need to prioritize the backend work

Comment (skylarbarrera) [reply src/redux/gas.ts]: added handling for error reporting, not for all gas related items if there is an error it never loads atm 😅 , we should revisit this after its handled via meterology(backend gas service) so we can retry etc.

Comment (skylarbarrera) [reply src/redux/gas.ts]: altho it will refetch every 2-3 seconds

File: src/redux/gas.ts [hunk 6]
```diff
@@ -498,16 +558,16 @@ export const gasPricesStartPolling = (
...
+                logger.debug('falling back to eth gas station');
               }
             }
             fetchResolve(true);
-          } catch (error) {
-            captureException(new Error('all gas estimates failed'));
-            logger.sentry('gas estimates error', error);
-            fetchReject(error);
+          } catch (e) {
+            logger.error(new RainbowError(`Gas Estimates Failed: ${e}`));
```
Comment (estrattonbailey) [inline src/redux/gas.ts:569]: Nit, best practice for errors: v descriptive messages. We might have other gas fetchers in the app, would be good to know that it was specifically Arbitrum, or even `getArbitrumGasPrices`, by including that in the message. Same for all these. Not a huge deal though.

**Review by estrattonbailey — APPROVED** [review]

**Review by estebanmino — APPROVED** [review]
didn't try it but lgtm

**Pull request merged**